tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/formatting2/TclFormatter.xtend

```diff
@@ -127,9 +128,12 @@ class TclFormatter extends XbaseFormatter {
 		macroTestStepContext.steps.forEach[format]
 	}
 
-	def dispatch void format(TestStep testStep, extension IFormattableDocument document) {
+	def dispatch void format(AbstractTestStep testStep, extension IFormattableDocument document) {
 		testStep.regionFor.keyword("-").prepend[newLine]
-		testStep.contents.forEach[format]
+		if (testStep instanceof TestStep) {
+			testStep.contents.forEach[format]
+			testStep.regionFor.keyword(".").prepend[noSpace]
```

Comment:
+1 for addressing this issue, but could you do this in a different pull request? Also for each formatter test an accompanying test should be written.